### PR TITLE
Replace `"--no-conda"` with `"--env-manager", "local"` in tests

### LINKED ISSUE #5590: Replace `["--no-conda"]` with `["--env-manager", "local"]` in tests
In https://github.com/mlflow/mlflow/pull/5567, we added `--env-manager` to specify an environment manager (e.g. `conda`) to restore a model's environment for MLflow Models and Projects, and marked `--no-conda` as deprecated. We still use `--no-conda` in tests, which should be replaced with `--env-manager local`.

### Example

https://github.com/mlflow/mlflow/blob/881919bc96f6858381e79f80c367f270395fb748/tests/catboost/test_catboost_model_export.py#L34

```diff
- EXTRA_PYFUNC_SERVING_TEST_ARGS = [] if _is_available_on_pypi("catboost") else ["--no-conda"] 
+ EXTRA_PYFUNC_SERVING_TEST_ARGS = [] if _is_available_on_pypi("catboost") else ["--env-manager", "local"] 
```

### Notes:

`--no-conda` in the following directories should not be replaced:

- `tests/projects`
- `tests/models`

### PR DESCRIPTION
**Pull request by nishipy**
## What changes are proposed in this pull request?

Fix #5590.

## How is this patch tested?

Existing tests

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@nishipy Thanks for the PR!

Can you also fix this line?

https://github.com/nishipy/mlflow/blob/9f6da780018a5b1a0f4acca7fd9142b9e41f6876/tests/diviner/test_diviner_model_export.py#L40

File: tests/helper_functions.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def pyfunc_serve_and_score_model(
     :param activity_polling_timeout_seconds: The amount of time, in seconds, to wait before
                                              declaring the scoring process to have failed.
     :param extra_args: A list of extra arguments to pass to the pyfunc scoring server command. For
-                       example, passing ``extra_args=["--no-conda"]`` will pass the ``--no-conda``
+                       example, passing ``extra_args=["--env-manager", "local"]`` will pass the ``--env-manager local``
```
Comment (harupy) [inline tests/helper_functions.py:161]: pylint complains this is too long

https://github.com/mlflow/mlflow/runs/5830151998?check_suite_focus=true#step:7:20

```
tests/helper_functions.py:161:0: C0301: Line too long (119/100) (line-too-long)
```

Can you adjust the line length by wrapping it?

**Comment by nishipy** [comment]
Hi @harupy,
Thank you for your points!
I will fix and commit them. Thanks

**Review by harupy — APPROVED** [review]
LGTM (if all the CI checks pass)!

**Comment by harupy** [comment]
@nishipy Thanks for working on this PR, would you be interested in making another contribution?

**Comment by nishipy** [comment]
> @nishipy Thanks for working on this PR, would you be interested in making another contribution?

@harupy Thanks for you your help. Yes, of course! I like it.

**Comment by harupy** [comment]
@nishipy Awesome, our team found this GitHub feature that annotates code with lint warnings/errors like the image below:

https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md#problem-matchers

image

In the attached image, javascript files are annotated. We want to do the same thing for python scripts.

Please let me know if you're interested in this.

**Comment by nishipy** [comment]
Actually it is the first time for me to see this GitHub feature, but it looks so interesting and useful for us. Let me know more details.
https://github.com/mlflow/mlflow/pull/5613/files

**Comment by harupy** [comment]
Of course, I described what needs to be done in my comment above.

**Comment by harupy** [comment]
@nishipy Let me know if you have questions.

**Comment by nishipy** [comment]
Thanks for the information. I will try it.
By the way, do we have to create a issue about that before submitting PR?

**Comment by harupy** [comment]
@nishipy No, we don't have to :)

**Comment by harupy** [comment]
@nishipy We can enable problem matchers like this:
https://github.com/ianchi/remotecodes/blob/cce29ae9846a759a4e518840458ced32275df3de/.github/workflows/ci.yml#L60

**Comment by nishipy** [comment]
@harupy Thanks! If I understand correctly, we have to implement the following things:
1. Add .json file to define regexp for detecting errors on pylint
2. Add the lines (as you describe just above) to the existing workflow definition

**Comment by harupy** [comment]
@nishipy Yes, exactly!

**Comment by nishipy** [comment]
Got it! I will start to do that later.

**Comment by nishipy** [comment]
Hi @harupy, I created PR #5639. Please take a look!

**Pull request merged**